src/esp/assets/ResourceManager.h

```diff
@@ -255,17 +255,52 @@ class ResourceManager {
                           scene::SceneNode* parent,
                           DrawableGroup* drawables);
 
-  // ======== Geometry helper functions ========
+  // ======== Geometry helper functions, data structures ========
 
   void translateMesh(BaseMesh* meshDataGL, Magnum::Vector3 translation);
 
   /**
-   * @brief Compute and return the axis aligned bounding box of a mesh.
+   * @brief Compute and return the axis aligned bounding box of a mesh in mesh
+   * local space
    * @param meshDataGL The mesh data.
    * @return The mesh bounding box.
    */
   Magnum::Range3D computeMeshBB(BaseMesh* meshDataGL);
 
+  /**
+   * @brief Compute the absolute AABBs for drawables in PTex mesh in world space
+   * @param baseMesh: ptex mesh
+   */
+
+  void computePTexMeshAbsoluteAABBs(BaseMesh& baseMesh);
+
+  /**
+   * @brief Compute the absolute AABBs for drawables in general mesh (e.g.,
+   * MP3D)world space
+   */
+  void computeGeneralMeshAbsoluteAABBs();
+
+  /*
+   * @brief Compute absolute transformations of all drwables stored in
+   * staticDrawableInfo_
+   */
+  std::vector<Magnum::Matrix4> computeAbsoluteTransformations();
+
+  // this helper vector contains all the drawables on which we will compute the
+  // absolute AABB
+  //
+  // pair: <Drawable, meshID>
+  //
+  // -) non-ptex mesh:
+  // meshID is the global index into meshes_.
+  // meshes_[meshID] is the BaseMesh corresponding to the drawable;
+  //
+  // -) ptex mesh:
+  // meshID is the index of the submesh corresponding to the drawable;
+  std::vector<std::pair<std::reference_wrapper<esp::gfx::Drawable>, uint32_t>>
```

Comment:
This vector is concerning.  Are we sure that we won't try to access the drawable after it gets deleted?

Reply:
This is a valid concern. However, it is unlikely happen.
This variable so far is used as a "temporary" variable that is to collect drawables, and meshIDs at the loading stage.  After that, it will not be used any more.

It is needed in multiple functions, so that I put it here as a member variable to avoid passing it as a function parameter.

I think solution is easy:
What I can do to address your concern is to clear this vector after its job is completed in loading stage.

Reply:
Is the drawable being used in anyway besides getting access to the node?  If so, why not store a reference to the scene node?  The scene node and drawable should have very similar lifetimes so it should be equally safe.

Reply:
Yes, good suggestion. In terms of functionality, both methods are the same. But I read my code, and what you suggested can potentially lead to simpler code. So why not? Let me try to change it.